### ISSUE TITLE
[Bug]: jest_config doesn't pass `data` for `js_library` target when snapshot updates are set

### ISSUE THREAD

**Issue by Aghassi** [aspect-build/rules_jest#128]
### What happened?

Our `jest_config` is exposed as a `js_library` target. Currently, the config was pretty vanilla, but we are adding a custom snapshot resolver. As such, we need an additional file to be passed along with the config. So, we added the `snapshotResolver.cjs` to the `js_library` as `data` since `jest_config` does not support taking more than one file as input.

This works when you run `bazel test`, the snapshots fail and state they need updating. However, if you run `bazel run`, the target can't find the snapshotResolver file. 

The workaround for now is for me to expose yet another `js_library` and pass that into the `data` attribute of the `jest_test` target, but this feels wrong because ideally dependencies of configs should be passed forward.

### Version

Development (host) and target OS/architectures:

Output of `bazel --version`: 6.1.0

Version of the Aspect rules, or other relevant rules from your
`WORKSPACE` or `MODULE.bazel` file:
rules_js: 1.25.0

Language(s) and/or frameworks involved:
Node 16.18.0

### How to reproduce

```shell
1. Have a macro that calls `jest_test` and passes a `js_library` target as the `jest_config` attribute. The library should be public visibility and include one index file as `srcs`.
2. Modify the `js_library` to have a `data` attribute and pass in a new snapshot resolver while updating the config to reference this file on disk
3. Run the macro to trigger the test. It will run fine for `bazel test`, but if you need to actually update snapshots the runner will fail since it can't find the resolver file.
```

**Comment by jbedard** [comment]
This is an issue in rules_js with how `js_library(data)` from the source dir doesn't get copied correctly when passed through `js_run_binary`.

https://github.com/aspect-build/rules_js/issues/1020 in rules_js would solve this use case, however https://github.com/aspect-build/rules_jest/issues/78 is probably the better solution for jest.

**Comment by jbedard** [comment]
Fixed in https://github.com/aspect-build/rules_js/commit/079a33dc7e5abaaf23681fc5af0f58a51fef1716, added jest tests in https://github.com/aspect-build/rules_jest/commit/da87af133ef3ce42c0e66fe52595816a6b0e47db
